pkg/cmd/admin/router/router.go

```diff
@@ -584,7 +581,19 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 			}
 		}
 
-		objects := []runtime.Object{
+		objects := []runtime.Object{}
+
+		if accountExists == false {
```

Comment:
I would expect the service account to be found or built later, and look more like this:
```
func generateServiceAccount(kClient *kclient.Client, ns string, name string, hostNetwork bool, secrets []*kapi.Secret) (serviceAccount *kapi.ServiceAccount, exists bool, needsSCC bool, needsUpdate bool, err error) {
  sa, err := kClient.ServiceAccounts(ns).Get(sa)
  if errors.IsNotFound(err) {
    sa = &kapi.ServiceAccount{ObjectMeta:kapi.ObjectMeta{Name:name}}
  } else if err != nil {
    return nil, false, false, err
  }

  if hostNetwork {
    ... check, set needsSCC ...
  }

  for _, secret := range secrets {
    ... ensure secret is referenced in sa, if not, add and set needsUpdate=true ...
  }

  return sa, exists, needsSCC, needsUpdate, nil
}
```

used like this:
```
secrets, ... = ...
objects = append(objects, secrets...)

serviceAccount, saExists, saNeedsSCC, saNeedsUpdate, err := generateServiceAccount(..., ..., secrets)
if !saExists {
  objects = append(objects, serviceAccount)
}

... bulk create

if saNeedsSCC {
  warn, including the command they'd need to run (`%parentName add-scc-to-user ... service-account-name`)
  use `serviceaccount.MakeUsername(namespace, name)` to build the service account name
}

if saNeedsUpdate {
  if _, err := kClient.ServiceAccounts(ns).Update(serviceAccount); err != nil {
    ... maybe just warn? we're not re-entrant at this point
  }
}

```